dotnet/src/SemanticKernel.Abstractions/Connectors/WebApi/Rest/RestApiOperationRunner.cs

```diff
@@ -84,6 +84,8 @@ public RestApiOperationRunner(HttpClient httpClient, AuthenticateRequestAsyncCal
             requestMessage.Content = payload;
         }
 
+        requestMessage.Headers.Add("User-Agent", "Microsoft-Semantic-Kernel");
```

Comment:
Could you add a comment on why we want to attach this header for *all* REST api operations?

Reply:
Made it optional -- RestApiOperationRunner now accepts the header value as an optional parameter of the class constructor allowing a client code to configure it if needed. The RestApiOperationRunner does not contain any SK SDK specific logic so far so this change is to keep it that way keep it that way.

Now this header is passed in KernelOpenApiExtensions with comment detailing requirement